composer/loggers/tensorboard_logger.py

```diff
@@ -0,0 +1,118 @@
+            the end of every epoch (`EVENT.EPOCH_END`), and the end of training
+            (`EVENT.FIT_END`). Default: ``100``.
+        rank_zero_only (bool, optional): Whether to log only on the rank-zero process.
+            Recommended to be true since the rank 0 will have access to most global metrics.
+            A setting of `False` may lead to logging of duplicate values.
+            Default: :attr:`True`.
+    """
+
+    def __init__(self,
+                 run_name: Optional[str] = None,
+                 log_dir: Optional[str] = None,
+                 flush_interval: int = 100,
+                 rank_zero_only: bool = True):
+
+        self.run_name = run_name
+        self.log_dir = log_dir
+        self.flush_interval = flush_interval
+        self.rank_zero_only = rank_zero_only
+        self.writer: SummaryWriter = SummaryWriter()
+
+    def log_data(self, state: State, log_level: LogLevel, data: Dict[str, Any]):
+        del log_level
+
+        if self.rank_zero_only and dist.get_global_rank() != 0:
+            return
+        for tag, data_point in data.items():
+            if isinstance(data_point, str):  # Will error out with weird caffe2 import error.
+                continue
+            # TODO: handle logging non-(scalars/arrays/tensors/strings)
+            # If a non-(scalars/arrays/tensors/strings) is passed, we skip logging it,
+            # so that we do not crash the job.
+            try:
+                self.writer.add_scalar(tag, data_point, global_step=int(state.timestamp.batch))
+            # Gets raised if data_point is not a tensor, array, scalar, or string.
+            except NotImplementedError:
+                pass
+
+    def init(self, state: State, logger: Logger) -> None:
+        if self.run_name is None:
+            self.run_name = state.run_name
+        if self.log_dir is None:
+            self.log_dir = 'tensorboard_logs'
+        summary_writer_log_dir = Path(self.log_dir) / self.run_name
+        self.writer = SummaryWriter(log_dir=summary_writer_log_dir, filename_suffix=self.run_name)
+
+    def batch_end(self, state: State, logger: Logger) -> None:
+        if int(state.timestamp.batch) % self.flush_interval == 0:
+            self._flush(logger)
+
+    def epoch_end(self, state: State, logger: Logger) -> None:
+        self._flush(logger)
+
+    def eval_end(self, state: State, logger: Logger) -> None:
+        self._flush(logger)
+
+    def fit_end(self, state: State, logger: Logger) -> None:
+        # Flush the file on fit_end, in case if was not flushed on epoch_end and the trainer is re-used
+        # (which would defer when `self.close()` would be invoked)
+        self._flush(logger)
+
+    def _flush(self, logger: Logger):
+        # To avoid empty log artifacts for each rank.
+        if self.rank_zero_only and dist.get_global_rank() != 0:
+            return
+        self.writer.flush()
+        file_path = self.log_dir  # Initialize for pyright.
```

Comment:
nit instead of the initialize and if check, add an assert is not None

```suggestion
        assert self.writer.file_writer is not None
        file_path = self.writer.file_writer.event_writer._file_name
```

Reply:
done